lxd-p2c static (Makefile) build fails

This is what happens when I use the lxd makefile to try to build lxd-p2c under Ubuntu 18.04. This would likely produce an amd64 binary though and thats not what I want.

This is the third ticket I've had to open for essentially the same issue. Please give verbose, newb friendly instructions on how I can cross compile lxd-p2c for i386. I can't see any 386 build options within the lxd Makefile and you said my previous method of specifying GOARCH=386 was incorrect so I really am out of ideas now.

```
~/go/src/github.com/lxc/lxd$ make lxd-p2c
CGO_ENABLED=0 go install -v -tags netgo ./lxd-p2c
github.com/lxc/lxd/shared/i18n
net
os/user
vendor/golang_org/x/net/lex/httplex
vendor/golang_org/x/net/proxy
github.com/google/uuid
net/textproto
crypto/x509
github.com/spf13/pflag
golang.org/x/sys/unix
crypto/tls
net/http/httptrace
net/http
net/http/cookiejar
github.com/gorilla/websocket
gopkg.in/macaroon-bakery.v2/httpbakery
github.com/lxc/lxd/shared
github.com/lxc/lxd/shared/osarch
github.com/lxc/lxd/shared/termios
github.com/lxc/lxd/shared/eagain
github.com/lxc/lxd/lxd/migration
github.com/lxc/lxd/shared/version
github.com/lxc/lxd/shared/simplestreams
github.com/lxc/lxd/client
github.com/lxc/lxd/lxc/utils
# github.com/lxc/lxd/lxc/utils
lxc/utils/progress.go:28:19: undefined: termios.GetSize
lxc/utils/progress.go:108:7: undefined: termios.IsTerminal
Makefile:30: recipe for target 'lxd-p2c' failed
make: *** [lxd-p2c] Error 2
```

--- a/shared/termios/termios_other.go
+++ b/shared/termios/termios_other.go
@@ -1,0 +1,32 @@
+// +build !windows,!cgo
+
+package termios
+
+import (
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// IsTerminal returns true if the given file descriptor is a terminal.
+func IsTerminal(fd int) bool {
+	return terminal.IsTerminal(fd)
+}
+
+// GetState returns the current state of a terminal which may be useful to restore the terminal after a signal.
+func GetState(fd int) (*terminal.State, error) {
+	return terminal.GetState(fd)
+}
+
+// GetSize returns the dimensions of the given terminal.
+func GetSize(fd int) (int, int, error) {
+	return terminal.GetSize(fd)
+}
+
+// MakeRaw put the terminal connected to the given file descriptor into raw mode and returns the previous state of the terminal so that it can be restored.
+func MakeRaw(fd int) (*terminal.State, error) {
+	return terminal.MakeRaw(fd)
+}
+
+// Restore restores the terminal connected to the given file descriptor to a previous state.
+func Restore(fd int, state *terminal.State) error {
+	return terminal.Restore(fd, state)
+}

